linux-openglow: Add OV5648 Module

Modifications to add OV5648 driver 4.9.xx kernel.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Scott Wiederhold <s.e.wiederhold@gmail.com>

diff -u a/drivers/media/platform/mxc/capture/Kconfig b/drivers/media/platform/mxc/capture/Kconfig
--- a/drivers/media/platform/mxc/capture/Kconfig
+++ b/drivers/media/platform/mxc/capture/Kconfig
@@ -73,6 +73,18 @@
 	depends on MXC_MIPI_CSI && I2C
 	---help---
 	  If you plan to use the ov5647 Camera with mipi interface in your MXC system, say Y here.
+
+config MXC_CAMERA_OV5648_MIPI
+        tristate "OmniVision ov5648 camera support using mipi"
+        depends on !VIDEO_MXC_EMMA_CAMERA && I2C
+        ---help---
+          If you plan to use the ov5648 Camera with mipi interface in your MXC system, say Y here.
+
+config MXC_CAMERA_SLAVE_OV5648_MIPI
+        tristate "OmniVision ov5648 slave dual camera support using mipi"
+        depends on !VIDEO_MXC_EMMA_CAMERA && I2C && MXC_CAMERA_OV5648_MIPI
+        ---help---
+          If you plan to use the second ov5648 Camera with mipi interface in your MXC system, say Y here.

 config MXC_VIDEO_GS2971
 	tristate "Gennum GS2971 SDI receiver"
diff -u a/drivers/media/platform/mxc/capture/Makefile b/drivers/media/platform/mxc/capture/Makefile
--- a/drivers/media/platform/mxc/capture/Makefile
+++ b/drivers/media/platform/mxc/capture/Makefile
@@ -37,6 +37,12 @@
 ov5647_camera_mipi-objs := ov5647_mipi.o
 obj-$(CONFIG_MXC_CAMERA_OV5647_MIPI) += ov5647_camera_mipi.o

+ov5648_camera_mipi-objs := ov5648_mipi.o
+obj-$(CONFIG_MXC_CAMERA_OV5648_MIPI) += ov5648_camera_mipi.o
+
+ov5648_camera_slave_mipi-objs := ov5648_slave_mipi.o
+obj-$(CONFIG_MXC_CAMERA_SLAVE_OV5648_MIPI) += ov5648_camera_slave_mipi.o
+
 tc358743_h2c_bridge-objs := tc358743_h2c.o
 obj-$(CONFIG_MXC_HDMI_CSI2_TC358743) += tc358743_h2c_bridge.o
 
